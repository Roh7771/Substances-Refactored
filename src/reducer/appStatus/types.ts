import { ModalWindowStatus, ErrorStatus } from '../../types';

export type AppStatusStateType = {
  modalWindowStatus: ModalWindowStatus,
  isSessionChecking: boolean,
  isRequestLoading: boolean,
  errorStatus: ErrorStatus
}

export type AppStatusActionConstsTypes = {
  SET_MODAL_WINDOW_STATUS: 'SET_MODAL_WINDOW_STATUS',
  SET_SESSION_CHECKING_STATUS: 'SET_SESSION_CHECKING_STATUS',
  SET_REQUEST_LOADING_STATUS: 'SET_REQUEST_LOADING_STATUS',
  SET_ERROR_STATUS: 'SET_ERROR_STATUS'
}

export type SetModalWindowStatusActionType = {
  type: AppStatusActionConstsTypes['SET_MODAL_WINDOW_STATUS'],
  payload: ModalWindowStatus
}

export type SetSessionCheckingStatusActionType = {
  type: AppStatusActionConstsTypes['SET_SESSION_CHECKING_STATUS'],
  payload: boolean
}

export type SetRequestLoadingStatusActionType = {
  type: AppStatusActionConstsTypes['SET_REQUEST_LOADING_STATUS'],
  payload: boolean
}

export type SetErrorStatusActionType = {
  type: AppStatusActionConstsTypes['SET_ERROR_STATUS'],
  payload: ErrorStatus
}

export type AppStatusActionTypes = SetModalWindowStatusActionType | SetErrorStatusActionType |
SetSessionCheckingStatusActionType | SetRequestLoadingStatusActionType
