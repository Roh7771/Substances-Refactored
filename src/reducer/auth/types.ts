import { AuthorizationStatus } from '../../const';
import { LoginDataType } from '../../types';

export type AuthStateType = {
  authorizationStatus: AuthorizationStatus;
  csrfToken: string;
};

export type AuthActionConstsTypes = {
  SET_AUTHORIZATION_STATUS: 'SET_AUTHORIZATION_STATUS';
  SET_CSRF_TOKEN: 'SET_CSRF_TOKEN';
  CHECK_SESSION: 'CHECK_SESSION';
  LOGIN: 'LOGIN';
};

export type SetAuthorizationStatusActionType = {
  type: AuthActionConstsTypes['SET_AUTHORIZATION_STATUS'];
  payload: AuthorizationStatus;
};

export type SetCSRFTokenActionType = {
  type: AuthActionConstsTypes['SET_CSRF_TOKEN'];
  payload: string;
};

export type CheckSessionActionType = {
  type: AuthActionConstsTypes['CHECK_SESSION'];
};

export type LoginActionType = {
  type: AuthActionConstsTypes['LOGIN'];
  payload: { loginData: LoginDataType; csrfToken: string };
};

export type AuthActionTypes =
  | SetAuthorizationStatusActionType
  | LoginActionType
  | SetCSRFTokenActionType
  | CheckSessionActionType;
