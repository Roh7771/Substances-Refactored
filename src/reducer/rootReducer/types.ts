import { AuthStateType, AuthActionTypes } from '../auth/types';
import { AppStatusStateType, AppStatusActionTypes } from '../appStatus/types';
import { SubstanceStateType, SubstanceActionTypes } from '../substance/types';

export type CombinedInitialState = {
  authState: AuthStateType,
  appStatusState: AppStatusStateType,
  substancesState: SubstanceStateType,
}

export type CombinedActionTypes = AuthActionTypes | AppStatusActionTypes | SubstanceActionTypes

export type RootReducerType = (state: CombinedInitialState, action: CombinedActionTypes) => CombinedInitialState
