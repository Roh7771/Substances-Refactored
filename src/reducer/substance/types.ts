import {
  LocationCollectionType,
  SubstanceType,
  QueryStringDataType,
} from '../../types';

export type SubstanceStateType = {
  substanceList: SubstanceType[];
  substanceToEdit: SubstanceType | null;
  locationCollection: LocationCollectionType;
  substancesToShow: number;
  queryStringData: QueryStringDataType;
};

export type SubstanceActionConstTypes = {
  SET_SUBSTANCE_LIST: 'SET_SUBSTANCE_LIST';
  SET_SUBSTANCE_TO_EDIT: 'SET_SUBSTANCE_TO_EDIT';
  SET_LOCATION_COLLECTION: 'SET_LOCATION_COLLECTION';
  ADD_SUBSTANCES_TO_SHOW_COUNT: 'ADD_SUBSTANCES_TO_SHOW_COUNT';
  RESET_SUBSTANCES_TO_SHOW_COUNT: 'RESET_SUBSTANCES_TO_SHOW_COUNT';
  SET_QUERY_SEARCH_DATA: 'SET_QUERY_SEARCH_DATA';
  SET_QUERY_LOCATION_DATA: 'SET_QUERY_LOCATION_DATA';
  LOAD_FULL_SUBSTANCE_LIST: 'LOAD_FULL_SUBSTANCE_LIST';
  QUERRY_NEW_DATA: 'QUERRY_NEW_DATA';
  CREATE_SUBSTANCE: 'CREATE_SUBSTANCE';
  UPDATE_SUBSTANCE: 'UPDATE_SUBSTANCE';
  DELETE_SUBSTANCE: 'DELETE_SUBSTANCE';
};

export type SetSubstanceListActionType = {
  type: SubstanceActionConstTypes['SET_SUBSTANCE_LIST'];
  payload: SubstanceType[];
};

export type SetSubstanceToEditActionType = {
  type: SubstanceActionConstTypes['SET_SUBSTANCE_TO_EDIT'];
  payload: SubstanceType | null;
};

export type SetLocationCollectionActionType = {
  type: SubstanceActionConstTypes['SET_LOCATION_COLLECTION'];
  payload: LocationCollectionType;
};

export type AddSubstanceToShowCountActionType = {
  type: SubstanceActionConstTypes['ADD_SUBSTANCES_TO_SHOW_COUNT'];
};

export type ResetSubstanceToShowCountActionType = {
  type: SubstanceActionConstTypes['RESET_SUBSTANCES_TO_SHOW_COUNT'];
};

export type SetQuerySearchDataActionType = {
  type: SubstanceActionConstTypes['SET_QUERY_SEARCH_DATA'];
  payload: {
    type: string;
    value: string;
  };
};

export type SetQueryLocationDataActionType = {
  type: SubstanceActionConstTypes['SET_QUERY_LOCATION_DATA'];
  payload: number[];
};

export type LoadFullSubstanceListActionType = {
  type: SubstanceActionConstTypes['LOAD_FULL_SUBSTANCE_LIST'];
};

export type QuerryNewDataActionType = {
  type: SubstanceActionConstTypes['QUERRY_NEW_DATA'];
  payload: QueryStringDataType;
};

export type CreateSubstanceActionType = {
  type: SubstanceActionConstTypes['CREATE_SUBSTANCE'];
  payload: {
    csrfToken: string;
    substance: SubstanceType;
    substanceList: SubstanceType[];
  };
};

export type UpdateSubstanceActionType = {
  type: SubstanceActionConstTypes['UPDATE_SUBSTANCE'];
  payload: {
    csrfToken: string;
    substance: SubstanceType;
    substanceList: SubstanceType[];
  };
};

export type DeleteSubstanceActionType = {
  type: SubstanceActionConstTypes['DELETE_SUBSTANCE'];
  payload: {
    csrfToken: string;
    substanceToDelete: SubstanceType;
    substanceList: SubstanceType[];
  };
};

export type SubstanceActionTypes =
  | SetSubstanceListActionType
  | SetSubstanceToEditActionType
  | SetLocationCollectionActionType
  | AddSubstanceToShowCountActionType
  | ResetSubstanceToShowCountActionType
  | SetQuerySearchDataActionType
  | SetQueryLocationDataActionType
  | LoadFullSubstanceListActionType
  | QuerryNewDataActionType
  | CreateSubstanceActionType
  | UpdateSubstanceActionType
  | DeleteSubstanceActionType;
