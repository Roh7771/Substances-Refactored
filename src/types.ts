export type SubstanceType = {
  name: string,
  amount?: string,
  casNumber?: string,
  location: number,
  place: string,
  number: number,
  _id?: number,
  company?: string
}

export type LoginDataType = {
  name: string,
  password: string
}

export type QueryStringDataType = {
  search: {
    type: string,
    value: string,
  },
  locations: number[]
}

export type LocationCollectionType = Map<number, Set<string>>

export type AllSubstancesServerResponseType = {
  substances: SubstanceType[];
};

export type CSRFTokenServerResponseType = {
  csrfToken: string;
};

export type OneSubstanceServerResponseType = {
  substance: SubstanceType;
};
