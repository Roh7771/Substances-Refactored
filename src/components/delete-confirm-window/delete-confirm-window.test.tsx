import * as React from 'react';
import * as renderer from 'react-test-renderer';
import DeleteConfirmWindow from './delete-confirm-window';
import { SubstanceType } from '../../types';

const noop = () => {};

const mockSubstanceList: SubstanceType[] = [
  {
    location: 111,
    name: 'some name',
    number: 1,
    place: 'some place',
    _id: 1,
  },
  {
    location: 111,
    name: 'some name',
    number: 1,
    place: 'some place',
    _id: 2,
  },
  {
    location: 111,
    name: 'some name',
    number: 1,
    place: 'some place',
    _id: 3,
  },
];

const mockSubstance: SubstanceType = {
  location: 111,
  name: 'some name',
  number: 1,
  place: 'some place',
  _id: 4,
};

it('Delete Confirm Window should render correctly', () => {
  const tree = renderer.create(
    <DeleteConfirmWindow
      dispatch={noop}
      substance={mockSubstance}
      substanceList={mockSubstanceList}
      resetModalWindow={noop}
      csrfToken="token"
    />,
  );
  expect(tree).toMatchSnapshot();
});
