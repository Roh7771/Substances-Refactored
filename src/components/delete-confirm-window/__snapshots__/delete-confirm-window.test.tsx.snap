// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete Confirm Window should render correctly 1`] = `
<div
  className="delete-confirm-field"
>
  <div
    className="delete-confirm-field__modal"
  >
    <h2
      className="delete-confirm-field__text"
    >
      Вы уверены, что хотите удалить этот реактив из базы данных?
    </h2>
    <div
      className="delete-confirm-field__buttons"
    >
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Да
      </button>
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Нет
      </button>
    </div>
  </div>
</div>
`;
